src/cluster_asm.h

```diff
@@ -13,8 +13,79 @@
+ * - On the destination side, Redis calls clusterAsmOnEvent(ASM_EVENT_IMPORT_COMPLETED)
+ *   when the import is completed.
+ * - Plugin calls clusterAsmRequest(ASM_REQUEST_CONFIG_UPDATED) to notify Redis
+ *   that the config is updated.
+ *
+ * Sequence diagram for import:
+ *   - Note: shows only the events that plugin needs to react.
+ *
+ * ┌───────────────┐              ┌───────────────┐         ┌───────────────┐             ┌───────────────┐
+ * │ Destination   │              │ Destination   │         │    Source     │             │ Source        │
+ * │ Cluster plugin│              │ Master        │         │    Master     │             │ Cluster plugin│
+ * └───────┬───────┘              └───────┬───────┘         └───────┬───────┘             └───────┬───────┘
+ *         │                              │                         │                             │
+ *         │ ASM_REQUEST_IMPORT_START     │                         │                             │
+ *         ├─────────────────────────────►│                         │                             │
+ *         │                              │CLUSTER SYNCSLOTS <arg>  │                             │
+ *         │                              ├────────────────────────►│                             │
+ *         │                              │                         │                             │
+ *         │                              │  SNAPSHOT(restore cmds) │                             │
+ *         │                              │◄────────────────────────┤                             │
+ *         │                              │  Repl stream            │                             │
+ *         │                              │◄────────────────────────┤                             │
+ *         │                              │                         │ ASM_EVENT_IMPORT_WAIT_PAUSE │
+ *         │                              │                         ├────────────────────────────►│
+ *         │                              │                         │  ASM_REQUEST_IMPORT_PAUSED  │
+ *         │                              │                         │◄────────────────────────────┤
+ *         │                              │ Drain repl stream       │                             │
+ *         │                              │◄────────────────────────┤                             │
+ *         │ASM_EVENT_IMPORT_WAIT_FINALIZE│                         │                             │
+ *         │◄─────────────────────────────┤                         │                             │
+ *         │                              │                         │                             │
+ *         │ ASM_REQUEST_CONFIG_UPDATED   │                         │                             │
+ *         ├─────────────────────────────►│                         │ ASM_REQUEST_CONFIG_UPDATED  │
```

Comment:
for the source cluster implementation/plugin, it sends this UPDATED message when it knows the topology is changed by gossip message or intenal cluster implementaion, right?

Reply:
i see, thank you